components/main-chef-wrapper/lib/version.go

```diff
@@ -0,0 +1,119 @@
+var manifestMap map[string]interface{}
+
+func init() {
+	gemManifestMap = gemManifestHash()
+	manifestMap = manifestHash()
+}
+func Version(){
+	if omnibusInstall() == true {
+		showVersionViaVersionManifest()
+	} else {
+		fmt.Fprintln(os.Stderr, "ERROR:", "Can not find omnibus installation directory for Chef Workstation")
+	}
+}
+
+func showVersionViaVersionManifest()  {
+	fmt.Printf("%v version: %v", dist.WorkstationProduct, componentVersion("build_version")  )
+	productMap := map[string]string{
+		dist.ClientProduct: dist.CLIWrapperExec,
+		dist.InspecProduct: dist.InspecCli,
+		dist.CliProduct: dist.CliGem,
+		dist.HabProduct: dist.HabSoftwareName,
+		"Test Kitchen": "test-kitchen",
+		"Cookstyle": "cookstyle",
+	}
+	for prodName, component := range productMap {
+		fmt.Printf("\n%v version: %v", prodName, componentVersion(component)  )
+	}
+	fmt.Printf("\n")
+}
+func componentVersion(component string) string  {
+	v, ok := gemManifestMap[component]
+	if ok {
+		stringifyVal := v.([]interface{})[0]
+		return stringifyVal.(string)
+	} else if v, ok := manifestMap[component]; ok {
+		return v.(string)
+	} else {
+		success, _ := Dig(manifestMap, "software", component, "locked_version")
+		if success == nil {
+			return "unknown"
+		}else {
+			return success.(string)
+		}
+	}
+}
+func gemManifestHash() map[string]interface{} {
+	filepath := path.Join(omnibusRoot(),"gem-version-manifest.json")
+	jsonFile, err := os.Open(filepath)
+	if err != nil {
+		fmt.Println(err)
+	}
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	var gemManifestHash map[string]interface{}
+	json.Unmarshal([]byte(byteValue), &gemManifestHash)
+	defer jsonFile.Close()
+	return gemManifestHash
+}
+func manifestHash() map[string]interface{}  {
+	filepath := path.Join(omnibusRoot(),"version-manifest.json")
+	jsonFile, err := os.Open(filepath)
+	if err != nil {
+		fmt.Println(err)
+	}
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	var manifestHash map[string]interface{}
+	json.Unmarshal([]byte(byteValue), &manifestHash)
+	defer jsonFile.Close()
+	return manifestHash
+}
+
+func omnibusInstall() bool {
+	//# We also check if the location we're running from (omnibus_root is relative to currently-running ruby)
+	//# includes the version manifest that omnibus packages ship with. If it doesn't, then we're running locally
+	//# or out of a gem - so not as an 'omnibus install'
+	ExpectedOmnibusRoot := ExpectedOmnibusRoot()
+	if _, err := os.Stat(ExpectedOmnibusRoot); err == nil {
+		if _, err = os.Stat(path.Join(ExpectedOmnibusRoot,"version-manifest.json")); err == nil {
+			return true
+		} else {
+			return false
+		}
+	} else {
+		return false
+	}
+}
+
+
+func omnibusRoot()string  {
+	omnibusroot, err := filepath.Abs(path.Join(ExpectedOmnibusRoot()))
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "ERROR:", "Can not find omnibus installation directory for Chef Workstation")
+	}
+	return omnibusroot
+}
+
+func ExpectedOmnibusRoot()string {
+	groot := os.Getenv("GEM_ROOT")
```

Comment:
I don't think we can assume this will be present in the user's environment because we're no longer loading this in the context of the ruby environment. 

  Instead, we should look relative to the `chef` executable directory.